src/thunderbots/software/ai/hl/stp/action/chip_action.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include "ai/hl/stp/action/action.h"
+#include "geom/angle.h"
+#include "geom/point.h"
+#include "shared/constants.h"
+
+/**
+ * The ChipAction makes the robot chip at the given location in the given
+ * direction, with the given power
+ */
+class ChipAction : public Action
+{
+   public:
+    /**
+     * Creates a new ChipAction
+     *
+     * @param length_of_region_behind_ball The distance behind the ball the robot must be
+     * before it can chip. The default value is 3 robot diameters. We want to keep the
+     * region small enough that we won't use the ChipIntent from too far away (since we
+     * risk colliding with something since the ChipIntent doesn't avoid obstacles), but
+     * large enough we can reasonable get in the region and chip the ball successfully.
```

Comment:
If we're not within this region, would another action (`MoveIntent`?) be called to move the robot to this region? Is it a good idea to implement this within the `ChipIntent`?

Reply:
@jessicahe21 	you're correct that if we're not in the region we will try move there first with a `MoveIntent`.

This precondition logic is not at the Intent level because Intents are more containers/datastructures than things that actually perform logic. It doesn't really make sense for a `KickIntent` to "call" a `MovePrimitive` in certain cases. More realistically this logic could be handled by the navigator, but IMO it makes more sense to keep it at a higher level, since this is more of a "logic" decision than a "path planning" decision.